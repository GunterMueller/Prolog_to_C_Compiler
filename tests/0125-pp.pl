:- include(library(pp)).

:- op(1100, xfx, '=>').
:- initialization op(1100, xfx, '=>'), main1.


term_expansion((X => Y), (rule :- (X) -> Y)).
term_expansion((X + Y), [declare(X/Y), add(X, Y)]).

goal_expansion(-X, \+X).

main1 :-
	see('tests/pp-sample.in'),
	main.
