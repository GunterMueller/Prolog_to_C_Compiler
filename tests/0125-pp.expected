rule :- a,\+b,c -> d,e.

rule :- x -> y.

foo(_1756,_1836) :- bar(_1756,_1796),_1796 = [baz|_1836].

declare(abc/def).

add(abc,def).

